### ISSUE TITLE
Support Application Tray

### ISSUE THREAD

**Issue by zeyugao** [microsoft/wslg#158]
When using Jetbrains ToolBox, I can shart it by command `jetbrains-toolbox` if no background process running. But if I minimize it, I can not get it show up again because it hides to the tray.

If I tap `jetbrains-toolbox` again, it will show up and disappear immediately.

**Comment by boltronics** [comment]
Implementing official support for running the full desktop experience (https://github.com/microsoft/wslg/issues/1019) could be a work-around for some people, depending on the use case.

I personally would like to see this implemented for Quod Libet, Pidgin and ownCloud (where I have to run it under WSL because there are files that are incompatible with the Windows filesystem).

**Comment by upintheairsheep** [comment]
> But the use cases are of course "infinite", it should be countless how many apps relies on the tray icon to function, especially the ones who closes to it as JetBrains Toolbox, Steam, Teams, Telegram, etc.

Yeah, technically it is infinite if you have an infinite amount of storage allowing for infinite permutations.

**Comment by Jackenmen** [comment]
As a workaround, it's possible to run one of the standalone system trays, e.g.
- [stalonetray](https://github.com/d3adb5/stalonetray)
- [trayer](https://github.com/sargon/trayer-srg)

If needed, such a system tray can be put as always on top (right above the Windows taskbar or wherever else, since these standalone trays are just a regular X11 borderless window) using some app that can set a Windows window to be always on top (my suggestion would be AHK script since you could probably get it to do it automatically on system startup then; however, PowerToys is also an option, at least if you disable the border effect it puts on a window).

I've also seen [gtk-sni-tray-standalone](https://github.com/taffybar/gtk-sni-tray) (or [taffybar](https://github.com/taffybar/taffybar) though that does more than just showing tray icons), but based on my testing, it doesn't work with weston (which would be consistent with this issue: https://github.com/taffybar/gtk-sni-tray/issues/33#issuecomment-1260044059)

The stalonetray and trayer don't support the newer StatusNotifierItem specification *but* that can be fixed by using a proxy that will add them as XEmbed tray icons - [snixembed](https://git.sr.ht/~steef/snixembed). Please note that you'll need to set `GDK_BACKEND=x11` env var when running the snixembed process (see https://todo.sr.ht/~steef/snixembed/17).

I imagine someone (ideally Microsoft ;)) could implement a Linux app taking advantage of [StatusNotifierWatcher](https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/StatusNotifierWatcher/) / [StatusNotifierHost](https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/StatusNotifierHost/) service and have it communicate with a Windows app that would then put icons in the Windows taskbar and allow interaction with them. Unlike the X11 System Tray Protocol spec, the StatusNotifierItem spec seems to provide generally useful data like the pixel data for the tray icon, title, and methods for interacting with the tray icon, rather than relying on X11's widget embedding protocol (even for the tray icon itself).
